dali/kernels/imgproc/resample/resampling_impl_cpu.h

```diff
@@ -83,6 +85,148 @@ void ResampleCol(Out *out, const In *in, int x, int w, const int32_t *in_columns
   }
 }
 
+template <typename Out, typename In>
+struct SIMD_vert_resample_impl {
+#ifdef __SSE2__
+  static constexpr int kVecSize = 16;
+  static constexpr int load_lanes = kVecSize / sizeof(In);
+  static constexpr int store_lanes = kVecSize / sizeof(Out);
+  static constexpr int kNumLanes = load_lanes > store_lanes ? load_lanes : store_lanes;
+  static constexpr int kNumVecs = kNumLanes * sizeof(float) / kVecSize;
+
+  using vec_pack = simd::multivec<kNumVecs>;
+#endif
+
+  static void run(Out *out, const In **rows, const float *kernel, int support,
+                   int begin_col, int end_col) {
+    int i = begin_col;
+#ifdef __SSE2__
+    for (; i + kNumLanes <= end_col; i += kNumLanes) {
+      vec_pack vtmp = vec_pack::zero();
+
+      for (int k = 0; k < support; k++) {
+        vec_pack vin = vec_pack::load(rows[k] + i);
+        __m128 coeff = _mm_set1_ps(kernel[k]);
+        for (int v = 0; v < kNumVecs; v++)
+          vtmp.v[v] = _mm_add_ps(vtmp.v[v], _mm_mul_ps(coeff, vin.v[v]));
+      }
+      store(out + i, vtmp);
+    }
+#endif
+    for (; i < end_col; i++) {
+      float tmp = 0;
+      for (int k = 0; k < support; k++)
+        tmp += rows[k][i] * kernel[k];
+      out[i] = ConvertSat<Out>(tmp);
+    }
+  }
+};
+
+
+template <typename Out, typename In>
+struct SIMD_horz_resample_impl {
+#ifdef __SSE2__
+  static constexpr int kVecSize = 16;
+  static constexpr int kNumLanes = kVecSize / sizeof(Out);
+  static constexpr int kNumVecs = kNumLanes * sizeof(float) / kVecSize;
+
+  using vec_pack = simd::multivec<kNumVecs>;
+#endif
+
+  template <int static_channels, bool clamp_left, bool clamp_right>
+  inline int run(Out *out, const In *in, int ox0, int ox1, int w,
+                 const int32_t *in_columns,
+                 const float *coeffs, int support,
+                 int dynamic_channels) {
+    const int channels = static_channels < 0 ? dynamic_channels : static_channels;
+
+    int x = ox0;
+#ifdef __SSE2__
+    float tmpin[kNumLanes];
+    for (; x + kNumLanes <= ox1; x += kNumLanes) {
+      Out tmp_out[kNumLanes];
+
+      if (static_channels < 0) {
+        // we don't know how many channels we have at compile time - inner loop over filter
+        for (int c = 0; c < channels; c++) {
+          vec_pack vout;
+          vout = vec_pack::zero();
+
+          for (int k = 0; k < support; k++) {
+            float tmp_coeffs[kNumLanes];
+            for (int l = 0; l < kNumLanes; l++)
+              tmp_coeffs[l] = coeffs[(x + l) * support + k];  // interleave per-column coefficients
+            vec_pack vcoeffs = vec_pack::load(tmp_coeffs);
+
+            for (int l = 0; l < kNumLanes; l++) {
+              int srcx = in_columns[x + l] + k;
+              if (clamp_left) if (srcx < 0) srcx = 0;
+              if (clamp_right) if (srcx > w-1) srcx = w-1;
+              tmpin[l] = in[srcx * channels + c];
```

Comment:
Does it make sense to make it a funciton?

Reply:
That's... risky. I don't want to go looking for the lost performance for having done so.

Reply:
Sure